lib/oci_utils/impl/oci-image-migrate-import-main.py

```diff
@@ -0,0 +1,256 @@
+        if bucketcontents is not None:
+            bucketcontentsdict = json.loads(bucketcontents)
+            _logger.debug(('Object storage contents: %s' % bucketcontentsdict))
+        else:
+            if verboseflag:
+                console_msg(msg='Bucket %s is emtpy.' % bucket)
+    except Exception as e:
+        exit_with_msg('Failed to get contents of %s: %s' % (bucket, str(e)))
+        #
+    # object exists?
+    if migrate_utils.object_exists(bucketcontents, objectname):
+        if verboseflag:
+            console_msg(msg='Object %s present in %s.' % (objectname, bucket))
+    else:
+        exit_with_msg('Object %s missing from %s.' % (objectname, bucket))
+    #
+    # display name exists?
+    cmd = ['oci', 'compute', 'image', 'list', '--compartment-id',
+           '%s' % compartment_id, '--display-name',
+           '%s' % displayname]
+    if verboseflag:
+        console_msg('Running: %s' % cmd)
+    objstat = migrate_tools.run_popen_cmd(cmd)
+    if objstat is not None:
+        exit_with_msg('Image with name %s already exists.' % displayname)
+    else:
+        if verboseflag:
+            console_msg('Image with name %s not yet imported.' % displayname)
+        else:
+            _logger.debug('Image with name %s not yet imported.' % displayname)
+    #
+    # upload?
+    if not read_yn('\n  Uploading %s to %s as %s'
+                   % (objectname, compartment, displayname)):
+        exit_with_msg('Exiting.\n')
+    cmd = ['oci', 'compute', 'image', 'import', 'from-object', '--bucket-name',
+           '%s' % bucket, '--compartment-id', '%s' % compartment_id,
+           '--name', '%s' % objectname, '--namespace', '%s' % namespace,
+           '--launch-mode', launchmode, '--display-name', '%s' % displayname]
+    if verboseflag:
+        console_msg(msg='Running: %s' % cmd)
+    try:
+        resval = migrate_tools.run_popen_cmd(cmd)
+    except Exception as e:
+        exit_with_msg('Failed to import %s: %s' % (objectname, str(e)))
+    #
+    # follow up
+    cmd = ['oci', 'compute', 'image', 'list', '--compartment-id',
+           '%s' % compartment_id, '--display-name', '%s' % displayname]
+    finished = False
+    _, clmns = os.popen('stty size', 'r').read().split()
+    importwait = migrate_tools.ProgressBar(int(clmns), 0.2,
+                                           progress_chars=['importing %s' %
+                                                           displayname])
+    importwait.start()
+    while not finished:
+        objstat = json.loads(migrate_tools.run_popen_cmd(cmd))
+        for ob in objstat['data']:
+            if ob['display-name'] == displayname:
+                if ob['lifecycle-state'] == 'AVAILABLE':
+                    finished = True
+            else:
+                console_msg(msg='%s not found.' % displayname)
+    if migrate_tools.isthreadrunning(importwait):
+        importwait.stop()
+    console_msg(msg='Done')
+    return 0
+
+
+if __name__ == "__main__":
```

Comment:
is this really needed ?

Reply:
it can take a very long time to import an image from object storage to custom images; the oci import command does return immediately, but the import is not finished.. avoiding people are going to restart the import in the first place, and to let them know when the import is done, and an instance can be created..